web/packages/teleport/src/Locks/NewLock.tsx

```diff
@@ -183,34 +183,37 @@ function TargetList({
   }
 
   const columns: TableColumn<any>[] = data.length
-    ? Object.keys(data[0]).map(c => {
-        const col: TableColumn<any> = {
-          key: c,
-          headerText: c,
-          isSortable: true,
-        };
-        if (c === 'labels') {
-          col.render = target => {
-            const labels = target.labels || [];
-            return (
-              <LabelCell data={labels.map(l => `${l.name}: ${l.value}`)} />
-            );
+    ? Object.keys(data[0])
+        .filter(k => k !== 'targetValue') // don't show targetValue in the table
+        .map(c => {
+          const col: TableColumn<any> = {
+            key: c,
+            headerText: c === 'lastUsed' ? 'Last Used' : c,
+            isSortable: true,
           };
-        }
-        return col;
-      })
+          if (c === 'labels') {
+            col.render = target => {
+              const labels = target.labels || [];
+              return (
+                <LabelCell data={labels.map(l => `${l.name}: ${l.value}`)} />
+              );
+            };
+          }
+          return col;
+        })
     : [];
 
   if (columns.length) {
     columns.push({
       altKey: 'add-btn',
-      render: ({ name }) => (
+      render: ({ targetValue }) => (
         <Cell align="right">
           <ButtonPrimary
-            onClick={onAdd.bind(null, name)}
+            onClick={onAdd.bind(null, targetValue)}
             data-testid="btn-cell"
             disabled={selectedLockTargets.some(
-              target => target.type === selectedTarget && target.name === name
+              target =>
+                target.type === selectedTarget && target.name === targetValue
```

Comment:
Won't this only work in cases where we set `targetValue` to `name`? IOW won't this only work for `user`/`node`/`windows_desktop`, but not for for `node`/`mfa_device`?

Reply:
I don't think so. What gives you that impression?

Reply:
I was thinking about it too, but it should be fine as `onAdd` sets `targetValue` as `target.name` now.

Reply:
Makes sense, I got myself confused about the data flow.